gui/default/syncthing/core/syncthingController.js

```diff
@@ -1047,6 +1047,28 @@ angular.module('syncthing.core')
             return deviceCfg.deviceID.substr(0, 6);
         };
 
+        $scope.getSupportBundle = function (){
+            $http.get(urlbase + '/system/support', {responseType: 'arraybuffer'})
```

Comment:
I think this thing is not required at all. A simple `<a href="...">` pointing directly to the `/system/support` endpoint should cause the browser to save the file, as you are setting the content-disposition header server-side. No javascript required. :)

Reply:
Because of using `<a href="...">` and access without a CSRF token, i changed the path `/rest/system/support` to '/rest/debug/support`. Like `cpuprof` and `heapprof`, I think `support bundle` should work when`debugging enabled`. @calmh ? @AudriusButkevicius ?

Reply:
I think not. Even with redacted passwords the support bundle contain too much private information to serve without auth. The existing debug stuff are just profiles, which don't contain anything sensitive.

I think linking directly should work anyway, because we use HTTP Basic auth for the login, with a session cookie, which the browser should supply by itself.

Reply:
I'm not sure understand u precisely. If I'm wrong please correct me. 
`handler` first goes to `csrfMiddleware(...)` function that use `/rest` as a prefix. In function we have 3 options. If begin with prefix `/rest` needs a api key, or if `rest/debug` needs neither a api key or a CSRF token. The third option is not begin with prefix `/rest` which needs a CSRF token, while don't have CSRF token http.Cookie() get a new one. 

So;
1. if `Support Bundle` is begin with prefix '/rest' and not '/rest/debug' we should serve a api key.
2. if begin with '/rest/debug' we should use debugging or new handleFunc like `getRestMux.HandleFunc("/rest/debug/support", s.getSupportBundle)` which don't need apikey, CSRF or debugging. **It is not a option as we agree.**
3. if **not** begin with prefix `/rest` needs a CSRF token. This time adding to `mux.Handle(...)` or `mux.HandleFunc(..)` like `s.getQR` is enough.

Reply:
OK OK, I was confused, sorry. The endpoints under `/rest/debug` require authentication as much as anything else, but indeed don't have the CSRF requirement. So yes, you are right, the best place for this is under `/rest/debug/support.zip` or something like that, and will require enabling debugging to get. I think that's an acceptable extra step.

Reply:
Thanks, I will push my code ASAP.